### PR TITLE
doc: Updates`mongodbatlas_advanced_cluster` ISS migration guide & resource doc with expected 500 error on update

### PR DESCRIPTION
**Pull request by maastha**
## Description

Updates`mongodbatlas_advanced_cluster` ISS migration guide & resource doc with expected 500 error on update.

Link to any related issue(s): [CLOUDP-269298](https://jira.mongodb.org/browse/CLOUDP-269298)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [X] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by erabil-mdb — APPROVED** [review]
LGTM!

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for following up with the doc changes

File: docs/guides/1.18.0-upgrade-guide.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ page_title: "Upgrade Guide 1.18.0"
 
 # MongoDB Atlas Provider 1.18.0: Upgrade and Information Guide
 
-***WARNING:*** For users using the `mongodbatlas_advanced_cluster` resource or data sources, and defining multi sharded clusters, this new version impacts the possibility of editing the definition of these clusters from the Atlas UI. This impact is limited to the first weeks of August 2024.
+***WARNING:*** For users using the `mongodbatlas_advanced_cluster` resource or data sources, and defining multi sharded clusters, this new version impacts the possibility of editing the definition of these clusters from the Atlas UI. This impact is limited to the first weeks of September 2024.
```
Comment (AgustinBettati) [inline docs/guides/1.18.0-upgrade-guide.md:7]: can you update the changelog entry that also points this out as well? As it is part of a version that is already released you can update the CHANGELOG.md file directly.

Comment (maastha) [reply docs/guides/1.18.0-upgrade-guide.md:7]: updated

**Pull request merged**